hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -580,7 +578,7 @@ private void handleInvocationException(Exception e) {
 
     private void notifyWithExceptionWhenTargetIsNull() {
         ClusterState clusterState = context.clusterService.getClusterState();
-        if (clusterState == FROZEN || clusterState == PASSIVE) {
+        if (!clusterState.isMigrationAllowed()) {
             notifyError(new IllegalStateException("Partitions can't be assigned since cluster-state: " + clusterState));
```

Comment:
Note : the check now includes `IN_TRANSITION` and `NO_MIGRATION`. 
On a side note, I'm confused with the exception message. "Partitions can't be assigned..." - but isn't this just an operation invocation where the target happened to be `null`?